### PR TITLE
Fixed: If the last message is deleted, the previous one becomes partially hidden #10765

### PR DESCRIPTION
**Pull request by advaitasol9**
@JmillsExpensify 
### Details

Deleting the last message of a conversation was resulting in the previous message getting partially/completely hidden because of extra scrolling, we have added required padding at the bottom of Flatlist to compensate the space required by 'recipient local time'.

### Fixed Issues

$ https://github.com/Expensify/App/issues/10765
$ https://github.com/Expensify/App/issues/10765#issuecomment-1254544138

### Tests

1. From home screen, start a chat with anyone.
2. Send 2-3 messages to the recipient.
3. Delete the last message.
4. After deleting the last message, the previous message should be visible and should not be partially or completely hidden.
- [x] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

1. From home screen, start a chat with anyone.
2. Send 2-3 messages to the recipient.
3. Delete the last message.
4. After deleting the last message, the previous message should be visible and should not be partially or completely hidden.
- [x] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/93445984/195361265-e877888f-ae88-487e-b77d-06d8fb04fd19.mov

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/93445984/195363006-b48e4fc7-7f8d-4318-be59-3e3eae5c4fef.mov

#### Mobile Web - Safari

https://user-images.githubusercontent.com/93445984/195361382-98321e41-cad4-45b6-a275-a6647067309a.mov

#### Desktop

https://user-images.githubusercontent.com/93445984/195361423-8a7ca95d-7f0b-4e43-b7a9-14f265aed36b.mov

#### iOS

https://user-images.githubusercontent.com/93445984/195362987-a00b5c90-689c-4f23-be19-3972fbbd3211.mov

#### Android

https://user-images.githubusercontent.com/93445984/195361526-012a46e5-acfd-48de-8ccd-1405015bed73.mov

### CODE REVIEW

**Comment by ctkochan22** [comment]
Looks okay to me, @Santhosh-Sellavel can you be a hero and test this PR?

**Comment by Santhosh-Sellavel** [comment]
@ctkochan22 Can you reproduce this issue now?

**Review by Santhosh-Sellavel — COMMENTED** [review]
Adds unnecessary spacing below the last message, if we want to make this change lets loop in the design team here, as this could be called a regression from a design point of view!

**Comment by ctkochan22** [comment]
Thats a good catch. We don't want that extra padding there

**Comment by JmillsExpensify** [comment]
@advaitasol9 are you planning on coming back to this PR and making updates this week? This one looks close to the finish line, let's get it merged!

**Comment by advaitasol9** [comment]
Yes @JmillsExpensify, will update in next few days.

**Comment by advaitasol9** [comment]
@Santhosh-Sellavel @JmillsExpensify  Updated

**Comment by advaitasol9** [comment]
Hi @JmillsExpensify @Santhosh-Sellavel please take a look here.

**Comment by Santhosh-Sellavel** [comment]
Oops, I forgot to submit the review!

**Comment by Santhosh-Sellavel** [comment]
Oops, I forgot to submit the review!
Screenshot 2022-11-03 at 1 49 52 AM

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ReportActionsList.js [hunk 1]
```diff
@@ -180,7 +180,7 @@ class ReportActionsList extends React.Component {
                     CellRendererComponent={this.renderCell}
                     contentContainerStyle={[
                         styles.chatContentScrollView,
-                        shouldShowReportRecipientLocalTime && styles.pt0,
+                        shouldShowReportRecipientLocalTime && styles.pt3 && styles.mtn4,
```
[the remark this answers is not in the document]
Comment (advaitasol9) [reply src/pages/home/report/ReportActionsList.js:183]: @Santhosh-Sellavel our original solution resulted in extra gap between the last message and 'Recipient Local time' as mentioned by you in previous comments, hence to counter the extra spacing we are providing negative margin, during testing i found -16 margin was perfect for our case.

[the remark this answers is not in the document]
Comment (Santhosh-Sellavel) [reply src/pages/home/report/ReportActionsList.js:183]: Why do we need to add a -16 margin, what's causing the issue?

[the remark this answers is not in the document]
Comment (advaitasol9) [reply src/pages/home/report/ReportActionsList.js:183]: In ios, the flatlist seems to scroll down more than required in absence of padding on rerendering with less items than before, hence we gave it a 12 padding first, but that padding caused to build a gap at bottom of list, to counter that gap we gave it a negative margin of 12 initially, but the negative margin of 12 still was off from the original design. Hence the -16 margin.

[the remark this answers is not in the document]
Comment (Santhosh-Sellavel) [reply src/pages/home/report/ReportActionsList.js:183]: @ctkochan22 Are we okay with this, can we loop in design to check if are we good with this?

**Comment by ctkochan22** [comment]
Trying to sort this out by testing now on main. And I can't reproduce. 

Can you guys help me out and confirm that the repro steps are
1. Send message 1
2. Send message 2
3. Delete most recent (message 2)
4. And it should jump above past message 1

**Comment by Santhosh-Sellavel** [comment]
Unable to reproduce this any more!

**Comment by Santhosh-Sellavel** [comment]
@advaitasol9 close this as it's no longer needed!

**Comment by ctkochan22** [comment]
Agreed that this isn't the right fix

**Pull request closed without merging**